ExecuteJobs.java

```diff
@@ -2,79 +2,109 @@
+      errorInfo.setErrorCode(ERR_FAILED_TO_KILL);
+      errorInfo.setIoMessage("Failed to kill " + pid);
+      return errorInfo;
     }
-    // Successfully killed pid
-    return 0;     
+
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("Successfully killed " + pid);
+    return errorInfo;     
   }
 
-  public String query(long pid) throws IOException {
-    Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
-    BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
-    String s = pInput.readLine();
-    if(s == null) {
-      return pid + " not found";
+  public ErrorInfo query(long pid) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+
+    ErrorInfo errorStatus = getProcessStatus(pid);
+    if(errorStatus.getIoMessage() == null) {
+      errorInfo.setErrorCode(ERR_FAILED_TO_FIND);
+      errorInfo.setIoMessage("Failed to find PID " + pid);
+      return errorInfo;
     }
-    return s;
+    errorInfo.setErrorCode(ERR_SUCCESS);
+    errorInfo.setIoMessage("PID " + pid + " status: " + errorStatus.getIoMessage());
+    return errorInfo;
   }
 
-  public ArrayList<String> currentJobStatus(ArrayList<Long> pids) throws IOException {
-    ArrayList<String> statuses = new ArrayList<String>();
+  public ArrayList<ErrorInfo> currentJobStatus(ArrayList<Long> pids) throws IOException {
+    ArrayList<ErrorInfo> statuses = new ArrayList<ErrorInfo>();
 
     int i = 0;
     while(i < pids.size()) {
       Long pid = pids.get(i);
-      Process p = Runtime.getRuntime().exec("ps -q " + pid + " -o state --no-headers");
-      BufferedReader pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
-      String status = pInput.readLine();
-      if(status == null) {
+      ErrorInfo errorStatus = getProcessStatus(pid);
+      ErrorInfo temp = new ErrorInfo();
+      if(errorStatus.getIoMessage() == null) {
         pids.remove(pid);
         continue;
       }
-      p = Runtime.getRuntime().exec("ps -p " + pid + " -o comm=");
-      pInput = new BufferedReader(new InputStreamReader(p.getInputStream()));
-      String name = pInput.readLine();
-      String result = name + " " + status + " " + pid;
-      statuses.add(result);
+      Process process = Runtime.getRuntime().exec("ps -p " + pid + " -o comm=");
+      BufferedReader processInput = new BufferedReader(new InputStreamReader(process.getInputStream()));
+      BufferedReader processError = new BufferedReader(new InputStreamReader(process.getErrorStream()));
+      temp.setErrorCode(ERR_SUCCESS);
+      temp.setErrorMessage(processError.readLine());
+      temp.setIoMessage(processInput.readLine());
+      statuses.add(temp);
       i++;
     }
 
     return statuses;
   }
+
+  public ErrorInfo getOutputOfRunningJob(long pid) throws IOException {
+    ErrorInfo errorInfo = new ErrorInfo();
+
+    Process process = Runtime.getRuntime().exec("ps " + pid);
```

Comment:
`ps <pid>` will not give you output of a running process AFAIK, for that you have to somehow capture output by redirecting file descriptors and then streaming the output.

Reply:
I believe I have fixed this now